File: realtime/ios/webrtc/Sources/Auth/BasicAuthProvider.swift
import Foundation

struct BasicAuthProvider: AuthProvider {
    var apiKey: String
    var apiBaseURL = URL(string: "https://api.inworld.ai")!

    func credentials() async throws -> RealtimeCredentials {
        let key = apiKey.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !key.isEmpty else { throw AuthError.missingAPIKey }
        return RealtimeCredentials(
            authorizationHeader: "Basic \(key)",
            apiBaseURL: apiBaseURL,
            callsURL: nil,
            preFetchedIceServers: nil
        )
    }
}

File: realtime/android/webrtc/app/src/main/java/ai/inworld/voiceagent/auth/BasicAuthProvider.kt
package ai.inworld.voiceagent.auth

/** The key from Settings is already base64 — sent verbatim after "Basic". */
class BasicAuthProvider(private val apiKey: String) : AuthProvider {
    override suspend fun credentials(): RealtimeCredentials {
        val key = apiKey.trim()
        if (key.isEmpty()) throw AuthException.MissingApiKey()
        return RealtimeCredentials(authorizationHeader = "Basic $key")
    }
}
